File: app/src/main/java/com/bitflaker/lucidsourcekit/general/database/values/SleepQuality.java
package com.bitflaker.lucidsourcekit.general.database.values;

public enum SleepQuality {
    Terrible("TRB"),
    Poor("POR"),
    Great("GRT"),
    Outstanding("OSD"),
    None("");

    private String id;

    SleepQuality(String id) {
        this.id = id;
    }

    public String getId() {
        return id;
    }

    public static SleepQuality getEnum(String id){
        for (SleepQuality enm : SleepQuality.values()) {
            if (enm.getId().equalsIgnoreCase(id)) {
                return enm;
            }
        }
        return null;
    }
}

File: app/src/main/java/com/bitflaker/lucidsourcekit/main/dreamjournal/rating/SleepQuality.kt
package com.bitflaker.lucidsourcekit.main.dreamjournal.rating

enum class SleepQuality(val id: String) {
    Terrible("TRB"),
    Poor("POR"),
    Great("GRT"),
    Outstanding("OSD"),
    None("");

    companion object {
        @JvmStatic
        fun getEnum(id: String) = SleepQuality.entries.find {
            it.id.equals(id, ignoreCase = true)
        }
    }
}
